File: 0112-path-sum/solution.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
void sumPath(struct TreeNode* root, int res,int *flag,int t){
    if(root == NULL) return ;
    if(!root->left && !root->right) {
    res += root->val;
    if(res==t){
        *flag = 1;
        return;
    }
    res = 0;
    }sumPath(root->left, res+root->val,flag,t);
    sumPath(root->right, res+root->val,flag,t);
}

bool hasPathSum(struct TreeNode* root, int t) {
    int flag = 0;
    sumPath(root,0,&flag,t);
    if(flag !=0) return true;
    return false;
}

File: 0112-path-sum/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private boolean dfs(TreeNode root, int tSum,int cSum){
        if(root == null) return false;
        cSum += root.val;
        if(root.left == null && root.right == null && tSum == cSum){
            return true;
        }
        return dfs(root.left, tSum, cSum) || dfs(root.right, tSum, cSum);
    }
    public boolean hasPathSum(TreeNode root, int targetSum) {
        return dfs(root,targetSum,0);
    }
}
